block/modes.go

```diff
@@ -75,3 +69,34 @@ func (m *Manager) runAsProposer(ctx context.Context, eg *errgroup.Group) error {
 
 	return nil
 }
+
+func (m *Manager) subscribeFullNodeEvents(ctx context.Context) {
+	if m.RunMode == RunModeProposer {
+		return
+	}
+	// Subscribe to new (or finalized) state updates events.
+	go uevent.MustSubscribe(ctx, m.Pubsub, "syncLoop", settlement.EventQueryNewSettlementBatchAccepted, m.onNewStateUpdate, m.logger)
+	go uevent.MustSubscribe(ctx, m.Pubsub, "validateLoop", settlement.EventQueryNewSettlementBatchFinalized, m.onNewStateUpdateFinalized, m.logger)
+
+	// Subscribe to P2P received blocks events (used for P2P syncing).
+	go uevent.MustSubscribe(ctx, m.Pubsub, "applyGossipedBlocksLoop", p2p.EventQueryNewGossipedBlock, m.OnReceivedBlock, m.logger)
+	go uevent.MustSubscribe(ctx, m.Pubsub, "applyBlockSyncBlocksLoop", p2p.EventQueryNewBlockSyncBlock, m.OnReceivedBlock, m.logger)
+}
+
+func (m *Manager) unsubscribeFullNodeEvents(ctx context.Context) {
+	if m.RunMode == RunModeProposer {
+		return
+	}
+
+	// unsubscribe for specific event (clientId)
+	unsubscribe := func(clientId string) {
+		err := m.Pubsub.UnsubscribeAll(ctx, clientId)
+		if err != nil {
+			m.logger.Error("Unsubscribe", "clientId", clientId, "error", err)
+		}
+	}
+	unsubscribe("syncLoop")
```

Comment:
hard-conding the client id can easily be error prone.. I'd suggest change to const.

Reply:
moved to const